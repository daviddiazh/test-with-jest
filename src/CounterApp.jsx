import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState(value);
    
    const handleAdd = (e) => {
        e.preventDefault();
        setCounter(counter + 1);
    }

    const handleSubstrack = (e) => {
        e.preventDefault();
        setCounter(counter - 1);
    }

    const handleReset = (e) => {
        e.preventDefault();
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <p>{counter}</p>
            <button
                onClick={ handleAdd }
            >
                + 1
            </button>
            <button
                onClick={ handleSubstrack }
            >
                - 1
            </button>
            <button
                onClick={ handleReset }
            >
                Reset
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 'El {value} no est√° definido'
}