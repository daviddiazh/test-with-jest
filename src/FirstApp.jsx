import PropTypes from 'prop-types';

export const FirstApp = ({ title, description }) => {

    return (
        <>
            <h1>{ title }</h1>
            {/* <p>{ description }</p> */}
        </>
    )

}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
}