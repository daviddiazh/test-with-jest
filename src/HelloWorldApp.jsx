import React from 'react'

const HelloWorldApp = () => {
    return (
        <div>Hola Mundo</div>
    )
}


export default HelloWorldApp