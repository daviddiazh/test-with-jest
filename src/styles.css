* {
    padding: 0px;
    margin: 0px;
}

body {
    background-color: #cecece;
    color: #333;
    font-family: sans-serif;
    text-align: center;
    line-height: 500%;
    margin: 5px 0;
}

button {
    padding: 10px 20px;
    margin: 0 10px;
    border: none;
    background-color: #1b74e4;
    color: #fff;
    cursor: pointer;
    border-radius: 7px;
}

p {
    font-size: 1.5em;
    margin: 0;
}