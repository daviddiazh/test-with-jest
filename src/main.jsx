import React from 'react'
import ReactDOM from 'react-dom/client'

import '../src/styles.css'  
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { TestLocation } from './TestLocation';


ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    {/* <FirstApp title='Hola, soy David' /> */}
    <CounterApp value={100} />
  </React.StrictMode>
)
