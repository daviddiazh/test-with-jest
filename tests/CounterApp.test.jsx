import { render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Pruebas de CounterApp', ()  => {

    const value = 100;

    test('Se debe hacer el match con el componente CounterApp', () => {

        const { container } = render( <CounterApp value={value} /> );

        expect( container ).toMatchSnapshot();

    });

    test('DeberÃ­a de mostrar el valor inicial de 100', () => {

        render( <CounterApp value={value} /> );
        expect( screen.getByText(100) ).toBeTruthy();

        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toBe('100')

    })

})