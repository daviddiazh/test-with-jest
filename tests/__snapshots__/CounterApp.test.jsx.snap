// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp Se debe hacer el match con el componente CounterApp 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    + 1
  </button>
  <button>
    - 1
  </button>
  <button>
    Reset
  </button>
</div>
`;
